### PR TITLE
feat(sdk): update to use cloudAdapter

### PR DESCRIPTION
**Pull request by yiqing-zhao**
E2E TEST: N/A

### CODE REVIEW

File: packages/sdk/src/conversationWithCloudAdapter/conversation.ts [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    };
+
+    return adapter;
+  }
+
+  /**
+   * The request handler to integrate with web request.
+   *
+   * @param req - An incoming HTTP [Request](xref:botbuilder.Request).
+   * @param req - The corresponding HTTP [Response](xref:botbuilder.Response).
```
Comment (tecton) [inline packages/sdk/src/conversationWithCloudAdapter/conversation.ts:180]: typo: `res`

File: packages/sdk/src/conversationWithCloudAdapter/sso/botSsoExecutionDialog.browser.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    authConfig: TeamsFx | OnBehalfOfCredentialAuthConfig,
+    ...args: any
+  ) {
+    throw new ErrorWithCode(
+      formatString(ErrorMessage.BrowserRuntimeNotSupported, "BotSsoExecutionDialog"),
+      ErrorCode.RuntimeNotSupported
+    );
+  }
+
+  // eslint-disable-next-line no-secrets/no-secrets
```
Comment (tecton) [inline packages/sdk/src/conversationWithCloudAdapter/sso/botSsoExecutionDialog.browser.ts:46]: Are these eslint-disable-* comments necessary? It looks like there's no secret next line.

File: packages/sdk/src/conversationWithCloudAdapter/middlewares/cardActionMiddleware.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import {
```
Comment (tecton) [inline packages/sdk/src/conversationWithCloudAdapter/middlewares/cardActionMiddleware.ts:1]: Files under `conversationWithCloudAdapter/middlewares` and `conversationWithCLoudAdapter/sso` are similar with those under `conversation/` except comments, do we need to keep the copy of these files?

File: packages/sdk/src/conversationWithCloudAdapter/sso/botSsoExecutionDialog.browser.ts (hunk 2 above)
Comment (yiqing-zhao) [reply packages/sdk/src/conversationWithCloudAdapter/sso/botSsoExecutionDialog.browser.ts:46]: image
Seems `BotSsoExecutionDialogHandler` is treated as secrets. Therefore, I add the eslint-disable-* comments. Same reason for other eslint-disable comments.

File: packages/sdk/src/conversationWithCloudAdapter/conversation.ts (hunk 1 above)
Comment (yiqing-zhao) [reply packages/sdk/src/conversationWithCloudAdapter/conversation.ts]: Thanks, updated!

File: packages/sdk/src/conversationWithCloudAdapter/middlewares/cardActionMiddleware.ts (hunk 3 above)
Comment (yiqing-zhao) [reply packages/sdk/src/conversationWithCloudAdapter/middlewares/cardActionMiddleware.ts]: Thanks, remove the duplicated files and redirect the references to the files under `conversation/`.

**Review by tecton — APPROVED** [review]

File: packages/sdk/src/conversationWithCloudAdapter/interface.ts [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+import { TeamsBotSsoPromptTokenResponse } from "../bot/teamsBotSsoPromptTokenResponse";
+import {
+  AuthenticationConfiguration,
+  OnBehalfOfCredentialAuthConfig,
+} from "../models/configuration";
+
+/**
+ * The response of a message action, e.g., `sendMessage`, `sendAdaptiveCard`.
+ */
+export interface MessageResponse {
```
Comment (swatDong) [inline packages/sdk/src/conversationWithCloudAdapter/interface.ts:23]: For developers, does new interface need to be `BotBuilderCloudAdapter.xxx` as well?
Any thoughts on deprecating old interfaces or reusing some interfaces?

File: packages/sdk/src/index.ts [hunk 5]
```diff
@@ -90,3 +90,6 @@ export {
   handleMessageExtensionQueryWithSSO,
 } from "./messageExtension/executeWithSSO";
 export { MessageExtensionTokenResponse } from "./messageExtension/teamsMsgExtTokenResponse";
+
+import * as BotBuilderCloudAdapter from "./conversationWithCloudAdapter/conversationWithCloudAdapter";
```
Comment (swatDong) [inline packages/sdk/src/index.ts:94]: Will we consider to make `BotBuilderCloudAdapter` the default one on next major version release?

File: packages/sdk/src/conversationWithCloudAdapter/interface.ts (hunk 4 above)
Comment (yiqing-zhao) [reply packages/sdk/src/conversationWithCloudAdapter/interface.ts]: Yes for the first question.
Deprecating and reusing some interfaces are good ideas. Just removed duplicated interfaces from `conversationWithCloudAdapter` and reuse interfaces from `conversation`. Old interfaces such as `NotificationOptions` and `ConversationOptions` are deprecated.
Thanks!

File: packages/sdk/src/index.ts (hunk 5 above)
Comment (yiqing-zhao) [reply packages/sdk/src/index.ts:94]: I propose to remove `BotBuilderCloudAdapter` on next major version release and migrate the changes back to `conversation` from `conversationWithCloudAdapter`.

**Review by swatDong — APPROVED** [review]

**Review by swatDong — DISMISSED** [review]

**Review by tecton — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Pull request merged**